### PR TITLE
[quality] add C/C++ header file nclude guard check

### PR DESCRIPTION
**Pull request by timothytrippel**
This adds a CI check for include guards in C/C++ header files. Additionally this fixes existing include guards in the code base to be style guide compliant.

### CODE REVIEW

File: util/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+exports_files([
```
Comment (cfrantz) [inline util/BUILD.bazel:5]: Add a comment:

TODO: make this a proper binary resource with `py_binary`.

File: util/fix_include_guard.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import argparse
+import re
+import sys
+from pathlib import Path
+
+PROJECT_NAME = "OPENTITAN_PROVISIONING"
+
+# This file is $REPO_TOP/util/fix_include_guard.py, so it takes two parent()
+# calls to get back to the top.
+REPO_TOP = Path(__file__).resolve().parent.parent
```
Comment (cfrantz) [inline util/fix_include_guard.py:28]: Other generalization TODOs:
Accept the project name as a cmdline parameter; plumb through to the rule attrs.
Accept the repo top as a cmdline parameter; use the `workspace` attr to learn the location of the repo top.

**Review by cfrantz — APPROVED** [review]
A couple of nits to improve long-term re-usability, but ok to commit as-is.

File: util/BUILD.bazel (hunk 1 above)
Comment (timothytrippel) [reply util/BUILD.bazel:6]: Done. Tracked in #41.

File: util/fix_include_guard.py (hunk 2 above)
Comment (timothytrippel) [reply util/fix_include_guard.py:29]: Done. Tracked in #41.

**Pull request merged**